### PR TITLE
Distinguish income from expense in cost center aggregate

### PR DESCRIPTION
**Pull request by mbayopanda**
This PR is related to #6023 and refactors some logic while having distinction between income and expense in the `cost_center_aggregate` table.

In this PR: 
- We have changed a bit the logic of `include_revenu` of  `Cost Center Allocation Report` 
- We adds the `is_income` column in the `cost_center_aggregate` table
- Update the procedure `zRecalculateCostCenterAggregates` to fill also the `is_income` column
- Update the procedure `ComputeCostCenterAllocationByIndex` to handle `is_income` column also, and fix data aggregation query

Benefit:
- Journal data are the same with data produced by the `Cost Center Allocation Report` (We have a way to compare data)

Todo:
- The client side of the `Cost Center Allocation Report` is still raw but we have to enhance that and fix some typos

Question:
- How does the `Cost Center Allocation Report` must be presented for revenue?
- Does for revenue also we can use the step-down algo? (with this PR we did that, you can explore that)

### CODE REVIEW

File: client/src/i18n/en/cost_center.json [hunk 1]
```diff
@@ -28,7 +28,9 @@
     "UPDATE_ALLOCATION_QUANTITIES_TOOLTIP": "Update system-computed cost center allocation quantities",
     "SET_ALLOCATION_QUANTITIES_TOOLTIP": "Define cost center allocation quantities",
     "INCLUDE_REVENUE_CENTERS" : "Include revenue with costs?",
-    "INCLUDE_REVENUE_CENTERS_HELP_TEXT" : "This will include the revenue generated by the revenue centers in addition to their costs."
+    "INCLUDE_REVENUE_CENTERS_HELP_TEXT" : "This will include the revenue generated by the revenue centers in addition to their costs.",
```
Comment (jmcameron) [inline client/src/i18n/en/cost_center.json:32]: I'm confused:  This is not currently used anywhere.   It seems like currently USE_REVENUE_CENTER* is used but not INCLUDE_REVENUE_CENTER*.  If INCLUDE_REVENUE_CENTER* are obselete, lets remove them.

File: server/models/admin.sql [hunk 2]
```diff
@@ -272,11 +272,18 @@ BEGIN
...
-    WHERE cost_center_id IS NOT NULL
-    GROUP BY cost_center_id, principal_center_id, period_id;
+  INSERT INTO cost_center_aggregate (period_id, credit, debit, cost_center_id, principal_center_id, is_income)
+    SELECT 
+      gl.period_id, 
+      SUM(gl.credit_equiv) AS credit, 
+      SUM(gl.debit_equiv) AS debit, 
+      gl.cost_center_id, 
+      gl.principal_center_id,
+      IF(a.type_id = 4, 1, 0) AS is_income
```
Comment (jmcameron) [inline server/models/admin.sql:282]: I did not realize you could do this.  I've been doing:   if (a.type_id IN (4, 1, 0))...

**Review by jmcameron — COMMENTED** [review]
Looks okay to me.   I followed most of the code/SQL changes.
I tested this with bhima_test and it seemed okay.  I looked at the new values in the cost_center_aggregate table and see the new column  (is_income) and it looks correct (although I could not be sure because there were no non-is_income entries.

**Comment by mbayopanda** [comment]
> 
> 
> Looks okay to me. I followed most of the code/SQL changes. I tested this with bhima_test and it seemed okay. I looked at the new values in the cost_center_aggregate table and see the new column (is_income) and it looks correct (although I could not be sure because there were no non-is_income entries.

Thank you @jmcameron for the review, with a production database you will see well the impact of `is_income` column in the `Cost Center Allocation Report`

**Review by jniles — APPROVED** [review]
Looks good to me.  As discussed offline, we need to make these both visible at the same time, in two different tables, as requested by Dr. Larry.

bors r+

**Comment by jniles** [comment]
@mbayopanda asked me to cancel this to fix up some typos.

**Comment by mbayopanda** [comment]
@jniles typo changes landed, for the expense and profit in the same report I suggest to do another PR for that

**Comment by jniles** [comment]
Bors r+

**Pull request merged**